### PR TITLE
executor: migrate test-infra to testify for executor/union_scan_test.go

### LINKED ISSUE #28623: migrate test-infra to testify for `union_scan_test.go`

### PR DESCRIPTION
**Pull request by tongtongyin**
### What problem does this PR solve?

Issue Number: close #28623

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/union_scan_test.go [hunk 1]
```diff
@@ -149,8 +153,11 @@ func (s *testSuite7) TestDirtyTransaction(c *C) {
 	tk.MustExec("commit;")
 }
 
-func (s *testSuite7) TestUnionScanWithCastCondition(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestUnionScanWithCastCondition(t *testing.T) {
```
Comment (hawkingrei) [inline executor/union_scan_test.go:156]: Please add ```t.Parallel()``` for every testing function.

Comment (tongtongyin) [reply executor/union_scan_test.go:158]: Has been added  /cc @hawkingrei

File: executor/union_scan_test.go [hunk 2]
```diff
@@ -214,8 +224,8 @@ func (s *testSuite7) TestUnionScanForMemBufferReader(c *C) {
 	tk.MustExec("insert t values (1,1),(2,2)")
 	tk.MustExec("begin")
 	_, err := tk.Exec("update t set b=b+1")
-	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[kv:1062]Duplicate entry '2' for key 'idx'")
+	require.NotNil(t, err)
+	require.Equal(t, "[kv:1062]Duplicate entry '2' for key 'idx'", err.Error())
```
Comment (hawkingrei) [inline executor/union_scan_test.go:228]: require.EqualError(t, err, "[kv:1062]Duplicate entry '2' for key 'idx'")

File: executor/union_scan_test.go [hunk 3]
```diff
@@ -330,14 +343,17 @@ func (s *testSuite7) TestForUpdateUntouchedIndex(c *C) {
 	tk.MustExec("create table t (a int,b int, unique index(a))")
 	tk.MustExec("begin")
 	_, err := tk.Exec("insert into t values (1, 1), (2, 2), (1, 3) on duplicate key update a = a + 1;")
-	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[kv:1062]Duplicate entry '2' for key 'a'")
+	require.NotNil(t, err)
+	require.Equal(t, "[kv:1062]Duplicate entry '2' for key 'a'", err.Error())
```
Comment (hawkingrei) [inline executor/union_scan_test.go:347]: ditto

Comment (tongtongyin) [reply executor/union_scan_test.go]: Done

**Review by hawkingrei — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]

**Pull request merged**